lib/charms/layer/containerd.py

```diff
@@ -0,0 +1,34 @@
+from charmhelpers.core import hookenv, host, unitdata
+
+
+def get_sandbox_image():
+    '''Return the container image location for the sandbox_image.
+
+    Set an appropriate sandbox image based on known registries. Precedence should be:
+    - related docker-registry
+    - default charmed k8s registry (if related to kubernetes)
+    - upstream
+
+    :return: str container image location
+    '''
+    # TODO: support k8s image-registry config (LP 1842958)
+    db = unitdata.kv()
+    canonical_registry = 'rocks.canonical.com:443/cdk'
+    upstream_registry = 'k8s.gcr.io'
+
+    docker_registry = db.get('registry', None)
+    if docker_registry:
+        sandbox_registry = docker_registry['url']
+    else:
+        try:
+            deployment = hookenv.goal_state()
+        except NotImplementedError:
+            deployment = {}
```

Comment:
If `goal-state` isn't available, should this fall back to `relation-list`?

Reply:
meh